File: HackerRank/Data Structures/Linked List/reverse.c
SinglyLinkedListNode* reverse(SinglyLinkedListNode* llist) 
{
    SinglyLinkedListNode* head = llist;
    SinglyLinkedListNode* prev = NULL; 
    while (head != NULL) {
        SinglyLinkedListNode* nextNode = head->next;
        head->next = prev;
        prev = head; 
        head = nextNode;
    }
    head = prev;
    return head;
}

File: HackerRank/Data Structures/Linked List/reverse.ts
function reverse(llist: DoublyLinkedListNode): DoublyLinkedListNode {
    let temp: DoublyLinkedListNode | null = null;
    let current: DoublyLinkedListNode | null = llist;
    while (current != null) {
        temp = current.prev;
        current.prev = current.next;
        current.next = temp;
        current = current.prev;
    }
    if (temp != null && temp.prev != null) llist = temp.prev;
    return llist;
}
